src/controller/CHIPDeviceController.cpp

```diff
@@ -328,16 +328,15 @@ CHIP_ERROR DeviceController::DisconnectDevice(NodeId nodeId)
 {
     ChipLogProgress(Controller, "Force close session for node 0x%" PRIx64, nodeId);
 
-    OperationalDeviceProxy * proxy = mSystemState->CASESessionMgr()->FindExistingSession(GetPeerScopedId(nodeId));
-    if (proxy == nullptr)
+    if (mSystemState->CASESessionMgr()->DisconnectSession(GetPeerScopedId(nodeId)))
```

Comment:
This is not right. Disconnecting a session previously meant marking the session as defunct. That is not what this method does.

Reply:
Marking the existing session as defunct is exactly what `mSystemState->CASESessionMgr()->DisconnectSession` is doing.

Reply:
Do you think we should remove `CASESessionManager::DisconnectSession` entirely and have `DeviceController::DisconnectDevice` use it's own reference to `mSystemState->SessionMgr()` to search for an mark as defunct the session itself

Reply:
> Marking the existing session as defunct 

That's only marking sessions that in the process of being established as defunct right? (which doesn't make a lot of sense either actually). This will not actually mark sessions that are already active as defunct because they won't be in the list of sessions tracked by `CASESessionManager` right?